overcooked_ai_py/mdp/overcooked_mdp.py

```diff
@@ -7,72 +7,390 @@
 
     def deepcopy(self):
-        return ObjectState(self.name, self.position, self.state)
+        return ObjectState(self.name, self.position)
 
     def __eq__(self, other):
         return isinstance(other, ObjectState) and \
             self.name == other.name and \
-            self.position == other.position and \
-            self.state == other.state
+            self.position == other.position
 
     def __hash__(self):
-        return hash((self.name, self.position, self.state))
+        return hash((self.name, self.position))
 
     def __repr__(self):
-        if self.state is None:
-            return '{}@{}'.format(self.name, self.position)
-        return '{}@{} with state {}'.format(
-            self.name, self.position, str(self.state))
+        return '{}@{}'.format(
+            self.name, self.position)
 
     def to_dict(self):
         return {
             "name": self.name,
-            "position": self.position,
-            "state": self.state
+            "position": self.position
         }
 
-    @staticmethod
-    def from_dict(obj_dict):
+    @classmethod
+    def from_dict(cls, obj_dict):
         obj_dict = copy.deepcopy(obj_dict)
         return ObjectState(**obj_dict)
 
 
+class SoupState(ObjectState):
+
+
+    def __init__(self, position, ingredients=[], cooking_tick=-1, **kwargs):
+        """
+        Represents a soup object. An object becomes a soup the instant it is placed in a pot. The
+        soup's recipe is a list of ingredient names used to create it. A soup's recipe is undetermined
+        until it has begun cooking. 
+
+        name (str): always "soup" for a soup instance
+        position (tupe): (x, y) coordinates in the grid
+        ingrdients (list(ObjectState)): Objects that have been used to cook this soup. Determiens @property recipe
+        cooking (int): How long the soup has been cooking for. -1 means cooking hasn't started yet
+        """
+        super(SoupState, self).__init__("soup", position)
+        self._ingredients = ingredients
+        self._cooking_tick = cooking_tick
+        self._recipe = None
+
+    def __eq__(self, other):
+        return isinstance(other, SoupState) and self.name == other.name and self.position == other.position and self._cooking_tick == other._cooking_tick and \
+            all([this_i == other_i for this_i, other_i in zip(self._ingredients, other._ingredients)])
+
+    def __hash__(self):
+        ingredient_hash = hash(tuple([hash(i) for i in self._ingredients]))
+        supercls_hash = super(SoupState, self).__hash__()
+        return hash((supercls_hash, self._cooking_tick, ingredient_hash))
+
+    def __repr__(self):
+        supercls_str = super(SoupState, self).__repr__()
+        ingredients_str = self._ingredients.__repr__()
+        return "{}\nIngredients:\t{}\nCooking Time:\t{}".format(supercls_str, ingredients_str, self._cooking_tick)
+
+    @property
+    def position(self):
+        return self._position
```

Comment:
I previously though this was necessary since I was overriding the `position.setter` but just discovered the python syntax to override only the setter. Hopefully this is cleaner now